agent/src/test/scala/za/co/absa/atum/agent/model/Measurement.scala

```diff
@@ -0,0 +1,109 @@
+package za.co.absa.atum.agent.model
+
+import org.apache.spark.sql.{SQLContext, SQLImplicits, SparkSession}
+import org.apache.spark.{SparkConf, SparkContext}
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.{BeforeAndAfterAll, BeforeAndAfterEach}
+import za.co.absa.atum.agent.AtumContext
+import za.co.absa.atum.agent.AtumContext.DatasetWrapper
+import za.co.absa.atum.agent.model.Measurement._
+
+class MeasurementSpec extends AnyFlatSpec with Matchers with BeforeAndAfterEach with BeforeAndAfterAll { self =>
+
+  @transient var ss: SparkSession = null
+  @transient var sc: SparkContext = null
+
+  private object testImplicits extends SQLImplicits {
+    protected override def _sqlContext: SQLContext = self.ss.sqlContext
+  }
+
+  override def beforeAll(): Unit = {
+    val sparkConfig = new SparkConf()
```

Comment:
You can use `SparkTestBase` from [spark-commons](https://github.com/AbsaOSS/spark-commons). It provides spark session in `spark` variable.